xdr/json.go

```diff
@@ -3,15 +3,62 @@ package xdr
 import (
 	"encoding/json"
 	"fmt"
+	"regexp"
+	"strconv"
 	"time"
+
+	"github.com/stellar/go/support/errors"
 )
 
+// ISO8601Time is a timestamp which supports parsing dates which have a year outside the 0000..9999 range
+type ISO8601Time struct {
+	time.Time
+}
+
+// reISO8601 is the regular expression used to parse date strings in the
+// ISO 8601 extended format, with or without an expanded year representation.
+var reISO8601 = regexp.MustCompile(`^([-+]?\d{4,})-(\d{2})-(\d{2})`)
+
+// MarshalJSON serializes the timestamp to a string
+func (t ISO8601Time) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf(`"%s"`, t.Format(time.RFC3339))), nil
```

Comment:
❗  To safely convert the string to JSON we should probably use `json.Marshal`. It might be safe as is, but I don't personally know all the character combinations that could end up in the string to know if it is safe. Given the json package is an expert in this we could rely on it for that part.

```suggestion
	return json.Marshal(t.Format(time.RFC3339))
```